.github/workflows/tests.yml

```diff
@@ -10,19 +10,15 @@ jobs:
   build:
     runs-on: ubuntu-latest
 
-    strategy:
-      matrix:
-        go: ['1.17', '1.16', '1.15', '1.14']
-
-    name: Go ${{ matrix.go }}
+    name: Unit
 
     steps:
     - uses: actions/checkout@v2
 
     - name: Set up Go
-      uses: actions/setup-go@v2
+      uses: actions/setup-go@v3
       with:
-        go-version: ${{ matrix.go }}
+        go-version-file: 'go.mod'
```

Comment:
can't hurt, can it? :)

Reply:
I'd prefer to keep the matrix and be explicit about which versions we're testing on, so that we can test on the latest two versions like Go does, for example 1.19 and 1.20 (when it comes out). Or is there a way to say "with go.mod version and later versions" or something?

Reply:
AFAIK there's no way to say "with go.mod version and later versions", other than specifying explicitly.

I don't really see why we need to test on multiple Go versions (Go should always build Pebble according to the version in `go.mod`). not gonna die on this hill though.

Reply:
As I see it, the reason is so that if we start using features only available in the latest version, people that are on the previous Go version don't get time to update their build scripts before their builds will start saying stuff like:

```
undefined: flag.TextVar
note: module requires Go 1.19
```

If we test on at least two versions, they at least get an additional 6 months to upgrade.

Go actually allows you to build modules that have later go versions in their go.mod, unless you use features from the newer Go version, of course, then it'll break.

Reply:
For a library, even just supporting two versions is fairly drastic, I think. At a previous company it'd take us a year or more to upgrade to the latest Go version.